### ISSUE TITLE
Change our Github Actions to only run relevant tests for the touched files

### ISSUE THREAD

**Issue by miaulalala** [nextcloud/mail#5205]
So only run PHP tests if a PHP file was touched, and so on - Spreed already does this:

https://github.com/nextcloud/spreed/blob/f97d7ab84fd3c0d87e46c9c08f3932ff57c5c3e7/tests/drone-run-php-tests.sh

**Comment by ChristophWurst** [comment]
is this still something we want?

**Comment by miaulalala** [comment]
It would reduce runtime considerably, even though our tests are fast  anyway. Plus it reduces our CO2 footprint.

**Comment by ChristophWurst** [comment]
Fair enough. We'll have to maintain a mapping of code type/path -> test suits though.

**Comment by miaulalala** [comment]
I'll fight you, no-response bot!
